specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,378 @@
+* NW sends the Post request to WNS’s NN which contains the persistent socket with the client
+* WNS’s NN maps the channel URL to the correct deviceId by decoding the deviceId from the channel URL and sends an NFY to the correct client.
+* WNS client receives the NFY and processes the channelId in the payload to retrieve the correct AppId from the database.
+* WNS client routes the notification to the appropriate app. If it needs to interface with the App, it uses BI, callbacks or WNF depending on the App Registration. If the payload is a toast, it gets directly routed to the Shell.
+
+## The problems today
+
+**Portal Registration**: The entire Registration flow for Push is dependent on registering an app in the partner centre portal by [associating an app with the Windows Store](https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-windows-store-dotnet-get-started-wns-push-notification). As a result, non-store provisioned apps like Win32 cannot avail of this feature even if they want to.
+
+**Client Registration**: Unpackaged 3rd party applications do not have a way today to register with the client because the API is private. The only way to get an implicit client side registration is to package the app as an MSIX or SSP which may not satisfy 3rd party requirements.
+
+**Push Channels**: All the Push APIs are dependent on the caller app having an identity at the minimal. However, even packaged Win32 apps that have an identity cannot use our Public APIs to subscribe to Push Channels because they may not be store provisioned. Sideloaded UWPs that aren't store distributed have exactly the same problem.
+
+**Activation**: When the native OS client receives a Push payload over the WNS socket, it signals the Background Infrastructure to trigger a task and run some code on the app's behalf. This flow is undefined for Win32 apps that need to use Push today mainly because of the Registration and API limitations described above. 
+
+**Channel Request Issues**: The channel APIs exposed today can throw exceptions via their Projections in the case of errors. Some errors can be critical errors while others are retryable errors. The msdn guidance for retrying isn't very clear and has been the root cause of multiple app bugs in the past. We want to abstract away the details of channel retry operations for developers in reunion.
+
+# Description
+
+At a high level, we need to provide a way for all Win32 applications to subscribe to and receive Push notifications irrespective of their app type. This includes unpackaged apps and packaged win32 (MSIX, Sparse Signed Packages). Moreover, all Push scenarios should adher to OS resource management policies like Power Saver, Network Attribution, Enterprise group policies etc. The Reunion Push component will abstract away the complexities of dealing with WNS channel registrations and Push related activations as much as possible freeing the developer to focus on other app related challenges. 
```

Comment:
I found a typo.
```suggestion
At a high level, we need to provide a way for all Win32 applications to subscribe to and receive Push notifications irrespective of their app type. This includes unpackaged apps and packaged win32 (MSIX, Sparse Signed Packages). Moreover, all Push scenarios should adhere to OS resource management policies like Power Saver, Network Attribution, Enterprise group policies etc. The Reunion Push component will abstract away the complexities of dealing with WNS channel registrations and Push related activations as much as possible freeing the developer to focus on other app related challenges. 
```

Reply:
Thanks for pointing the typo. Will make the change.